src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.cs

```diff
@@ -91,14 +101,70 @@ protected virtual SyntaxToken GetAdjustedTokenForPragmaRestore(SyntaxToken token
             return token;
         }
 
-        public async Task<IEnumerable<CodeFix>> GetSuppressionsAsync(Document document, TextSpan span, IEnumerable<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        public Task<IEnumerable<CodeFix>> GetSuppressionsAsync(Document document, TextSpan span, IEnumerable<Diagnostic> diagnostics, CancellationToken cancellationToken)
         {
-            var suppressableDiagnostics = diagnostics.Where(CanBeSuppressed);
-            if (suppressableDiagnostics.IsEmpty())
+            return GetSuppressionsAsync(document, span, diagnostics, skipSuppressMessage: false, cancellationToken: cancellationToken);
+        }
+
+        internal async Task<IEnumerable<PragmaWarningCodeAction>> GetPragmaSuppressionsAsync(Document document, TextSpan span, IEnumerable<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            var codeFixes = await GetSuppressionsAsync(document, span, diagnostics, skipSuppressMessage: true, cancellationToken: cancellationToken).ConfigureAwait(false);
+            return codeFixes.SelectMany(fix => ((SuppressionCodeAction)fix.Action).NestedActions).Cast<PragmaWarningCodeAction>();
+        }
+
+        private async Task<IEnumerable<CodeFix>> GetSuppressionsAsync(Document document, TextSpan span, IEnumerable<Diagnostic> diagnostics, bool skipSuppressMessage, CancellationToken cancellationToken)
+        {
+            // We only care about diagnostics that can be suppressed.
+            diagnostics = diagnostics.Where(CanBeSuppressed);
+            if (diagnostics.IsEmpty())
             {
-                return null;
+                return SpecializedCollections.EmptyEnumerable<CodeFix>();
+            }
+
+            var suppressionTargetInfo = await GetSuppressionTargetInfoAsync(document, span, cancellationToken).ConfigureAwait(false);
+            if (suppressionTargetInfo == null)
+            {
+                return SpecializedCollections.EmptyEnumerable<CodeFix>();
+            }
+
+            if (!skipSuppressMessage)
+            {
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var suppressMessageAttribute = semanticModel.Compilation.SuppressMessageAttributeType();
+                skipSuppressMessage = suppressMessageAttribute == null || !suppressMessageAttribute.IsAttribute();
```

Comment:
Did we really find a case where there can be a type named SuppressMessageAttribute that is not a attribute? Just curious :)